### PR TITLE
[W-8840996] Outbound Funds NPSP: Add automated checks + dev tools

### PR DESCRIPTION
**Pull request by spelak-salesforce**
- Adds dev tools + automated Github Action Pull Request checks.
    - cspell
    - prettier
    - eslint
    - lockfile-lint
    - string label audit
    - flake8 and black for python linting and formatting
- Upgraded to API Version `50.0` to keep this repo up to date
- Had to fix all prettier, eslint, spelling, string label audit violations.  No functionality should have changed.
- Added a(n incomplete) Jest test so the the Jest Test Github Action runs successfully (it needs at least one test).
- Added a sample python test so the Python / pytest Github Action runs successfully (it needs at least one test).
- Added storytelling data.

# Critical Changes

# Changes

# Issues Closed

# Unaggregated Pull Requests
* Updating formatting to meet standards part 1 [[PR27](https://github.com/SalesforceFoundation/OutboundFundsNPSP/pull/27)]

### CODE REVIEW

**Comment by spelak-salesforce** [comment]
@davidmreed , I want to add a PR that installs our team's pre-commit hooks, make minimal changes to code so they pass the linting and formatting checks, and upgrade to API Version `50.0`.

This will be a massive Pull Request, and I am trying to make reviewing this Pull Request as digestible as possible.   This PR only includes adding our team's pre-commit hooks so far.  Does it look valid?  

Child PR #27 will include the code changes + other changes needed to run on API Version `50.0`.

File: .prettierignore [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+*.robot
+*.pdf
+*.design
+
+# Filetypes to ignore
+*.sql
+*.xml
+
+# Robot
+robot/Grants-Management/results
```
Comment (davidmreed) [inline .prettierignore:32]: This ignore line needs to be revised

File: package.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{
+  "name": "OutboundFundsNPSP",
+  "license": "UNLICENSED",
+  "author": "Salesforce.org Impact Engineering",
+  "devDependencies": {
+    "@lwc/eslint-plugin-lwc": "^0.10.0",
+    "@lwc/jest-preset": "^7.0.0",
+    "@sa11y/jest": "^0.2.0",
+    "@salesforce/eslint-config-lwc": "^0.6.0",
+    "@salesforce/lwc-jest": "^0.6.2",
```
Comment (davidmreed) [inline package.json:10]: Is this the right version of lwc-jest for API 50? I thought this was older but I may be misremembering.

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Grants Management
```
Comment (davidmreed) [inline requirements.txt:1]:
```suggestion
# Outbound Funds NPSP
```

**Review by davidmreed — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (spelak-salesforce) [reply package.json:10]: It seems there was an error creating new versions of `@salesforce/sfdx-lwc-jest`.   API Version 50.0 uses `0.11.0` while API Version 51.0 is on `0.10.x`.   Updating.

**Comment by spelak-salesforce** [comment]
@davidmreed , made requested changes + added Github actions from [GMS](https://github.com/SalesforceFoundation/Grants-Management).

Note: Packaged code changes are in the child branch #27

**Review by davidmreed — APPROVED** [review]

**Comment by spelak-salesforce** [comment]
@davidmreed , does our [Python / pytest check fail](https://github.com/SalesforceFoundation/OutboundFundsNPSP/pull/26/checks?check_run_id=1817331671) because we don't have any python in the repo?   Should we remove this Github action?

**Comment by spelak-salesforce** [comment]
[W-8840996](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B00000095eIQIAY/view)

**Comment by davidmreed** [comment]
> @davidmreed , does our [Python / pytest check fail](https://github.com/SalesforceFoundation/OutboundFundsNPSP/pull/26/checks?check_run_id=1817331671) because we don't have any python in the repo? Should we remove this Github action?

Yeah. If you're not planning to use any Python tasks, feel free to remove it.

**Comment by spelak-salesforce** [comment]
> > @davidmreed , does our [Python / pytest check fail](https://github.com/SalesforceFoundation/OutboundFundsNPSP/pull/26/checks?check_run_id=1817331671) because we don't have any python in the repo? Should we remove this Github action?
> 
> Yeah. If you're not planning to use any Python tasks, feel free to remove it.

@davidmreed , I prefer keeping the Github Action to protect any future python added to the repo.  I added `tasks/tests/test_sample.py` as a sample test since the Github Action requires at least one python test.   If you don't like this, can you re-review with changes requested?  Thank you!

File: force-app/main/default/flexipages/Disbursement_NPSP_Record.flexipage-meta.xml [hunk 4]
```diff
@@ -1,191 +1,219 @@
...
-                <name>collapsed</name>
-                <value>false</value>
-            </componentInstanceProperties>
-            <componentInstanceProperties>
-                <name>hideChatterActions</name>
-                <value>false</value>
-            </componentInstanceProperties>
-            <componentName>force:highlightsPanel</componentName>
-        </componentInstances>
+        <itemInstances>
```
Comment (spelak-salesforce) [inline force-app/main/default/flexipages/Disbursement_NPSP_Record.flexipage-meta.xml:4]: `itemInstances` is API Version 49.0+ markup.

File: force-app/main/default/lwc/lookup/lookup.js [hunk 5]
```diff
@@ -1,287 +1,320 @@
...
+        },
+        multiEntry: {
+            ariaLabel: MULTI_ENTRY_ARIA_LABEL,
+        },
+    };
+
+    _selection = [];
+
+    @api
+    get selection() {
```
Comment (spelak-salesforce) [inline force-app/main/default/lwc/lookup/lookup.js:38]: selection was originally being modified and lwc/eslint didn't like an `@api` being directly modified.  Change to having `@api` getter/setter modify a private variable.

**Review by scottwarren-sfdo — APPROVED** [review]
Looks great. Thanks for taking this on!

**Pull request merged**